cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/StreamFileJanitor.java

```diff
@@ -63,25 +70,48 @@ public void cleanAll() throws Exception {
     List<NamespaceMeta> namespaces = namespaceQueryAdmin.list();
 
     for (NamespaceMeta namespace : namespaces) {
-      Location streamBaseLocation =
+      final Location streamBaseLocation =
         namespacedLocationFactory.get(Id.Namespace.from(namespace.getName())).append(streamBaseDirPath);
-      if (!streamBaseLocation.exists()) {
+      UserGroupInformation ugi = impersonator.getUGI(new NamespaceId(namespace.getName()));
+      boolean exists = ImpersonationUtils.doAs(ugi, new Callable<Boolean>() {
```

Comment:
I want to get the ugi and use it for multiple operations. Hence I am using the doAs method in ImpersonationUtils.